Std/Data/List/Basic.lean

```diff
@@ -685,6 +685,20 @@ partitionMap (id : Nat ⊕ Nat → Nat ⊕ Nat) [inl 0, inr 1, inl 2] = ([0, 2],
     | .inl a => go xs (acc₁.push a) acc₂
     | .inr b => go xs acc₁ (acc₂.push b)
 
+/-- Monadic generalization of `List.partition`. -/
+@[inline] def partitionM [Monad m] (p : α → m Bool) (l : List α) : m (List α × List α) :=
+  go l #[] #[] where
```

Comment:
```suggestion
  go l #[] #[]
where
```
The convention here is to put `where` on its own line, unless it can fit on the same line as the body and the `:=`